index.js

```diff
@@ -76,7 +80,8 @@ module.exports = fp(function from (fastify, opts, next) {
         // detect if body should be encoded as JSON
         // supporting extended content-type header formats:
         // - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type
-        const shouldEncodeJSON = contentType.toLowerCase().indexOf('application/json') === 0
+        const plainContentType = contentType.toLowerCase().split(';')[0]
```

Comment:
Please avoid the use of split() as it allocates a new array. You can use `str.slice(0, str.indexOf(':'))` instead